lib/integrations/awsoidc/deployservice_iam_config.go

```diff
@@ -0,0 +1,328 @@
+		}
+		return trace.Wrap(err)
+	}
+
+	log.Printf("TaskRole: Boundary Policy %q created.\n", req.TaskRoleBoundaryPolicyName)
+	return nil
+}
+
+// createTaskRole creates the TaskRole and sets up the Role Boundary and its Trust Relationship.
+func createTaskRole(ctx context.Context, clt DeployServiceIAMConfigureClient, req DeployServiceIAMConfigureRequest) error {
+	taskRoleAssumeRoleDocument, err := awslib.NewPolicyDocument(
+		awslib.StatementForECSTasksAssumeRole(),
+	).Marshal()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	policyARNForRoleBoundary := awslib.PolicyARN(req.partitionID, req.AccountID, req.TaskRoleBoundaryPolicyName)
+
+	_, err = clt.CreateRole(ctx, &iam.CreateRoleInput{
+		RoleName:                 &req.TaskRole,
+		AssumeRolePolicyDocument: &taskRoleAssumeRoleDocument,
+		PermissionsBoundary:      &policyARNForRoleBoundary,
+		Tags:                     req.ResourceCreationTags.ToIAMTags(),
+	})
+	if err != nil {
+		if trace.IsAlreadyExists(awslib.ConvertIAMv2Error(err)) {
+			return trace.AlreadyExists("Role %q already exists, please remove it and try again.", req.TaskRole)
+		}
+		return trace.Wrap(err)
+	}
+
+	log.Printf("TaskRole: Role %q created with Boundary %q.\n", req.TaskRole, policyARNForRoleBoundary)
+	return nil
+}
+
+// addPolicyToTaskRole updates the TaskRole to allow the service to:
+// - manage Policies of the TaskRole
+// - write logs to CloudWatch
+func addPolicyToTaskRole(ctx context.Context, clt DeployServiceIAMConfigureClient, req DeployServiceIAMConfigureRequest) error {
+	taskRoleARN := awslib.RoleARN(req.partitionID, req.AccountID, req.TaskRole)
+
+	taskRolePolicyDocument, err := awslib.NewPolicyDocument(
+		awslib.StatementForIAMEditRolePolicy(taskRoleARN),
+		awslib.StatementForWritingLogs(),
+	).Marshal()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	_, err = clt.PutRolePolicy(ctx, &iam.PutRolePolicyInput{
+		PolicyName:     &req.TaskRole,
+		RoleName:       &req.TaskRole,
+		PolicyDocument: &taskRolePolicyDocument,
+	})
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	log.Printf("TaskRole: IAM Policy %q added to Role %q.\n", req.TaskRole, req.TaskRole)
+	return nil
+}
+
+// addPolicyToIntegrationRole creates or updates the DeployService Policy in IntegrationRole.
+// It allows the Proxy to call ECS APIs and to pass the TaskRole when deploying a service.
+func addPolicyToIntegrationRole(ctx context.Context, clt DeployServiceIAMConfigureClient, req DeployServiceIAMConfigureRequest) error {
+	taskRoleARN := awslib.RoleARN(req.partitionID, req.AccountID, req.TaskRole)
+
+	taskRolePolicyDocument, err := awslib.NewPolicyDocument(
+		awslib.StatementForIAMPassRole(taskRoleARN),
+		awslib.StatementForECSManageService(),
+	).Marshal()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	_, err = clt.PutRolePolicy(ctx, &iam.PutRolePolicyInput{
+		PolicyName:     &req.IntegrationRoleDeployServicePolicy,
```

Comment:
Can an intergration role used for multiple task roles? if so, the policy names will collide.

Reply:
We might need to change this when we have multiple Deployment Modes.
For now, there's only `database-service` so they should not collide.

I'll make sure to add a suffix when we add a policy for another deployment.
Or we might have this policy always having the set of required statements for the deployments we have